docs/design-docs/08-package-variant.md

```diff
@@ -640,19 +707,530 @@ understanding, we can see that the controller will perform mutations on the
+| cluster-04 | foo-a        |
+| cluster-04 | foo-b        |
+
+All of those PackageVariants have the same upstream.
+
+The second criteria targeting is via a label selector against Porch Repository
+objects, along with a list of package names. Those packages will be instantiated
+in each matching repository. Just like in the first example, not listing a
+package name defaults to one package, with the same name as the upstream
+package. Suppose, for example, we have these four repositories defined in our
+Porch cluster:
+
+| Repository | Labels                                |
+| ---------- | ------------------------------------- |
+| cluster-01 | region=useast1, env=prod, org=hr      |
+| cluster-02 | region=uswest1, env=prod, org=finance |
+| cluster-03 | region=useast2, env=prod, org=hr      |
+| cluster-04 | region=uswest1, env=prod, org=hr      |
+
+If we create a PackageVariantSet with the following `spec`:
+
+```yaml
+spec:
+  upstream:
+    repo: example-repo
+    package: foo
+    revision: v1
+  targets:
+  - repositorySelector:
+      matchLabels:
+        env: prod
+        org: hr
+  - repositorySelector:
+      matchLabels:
+        region: uswest1
+      packageNames:
+      - foo-a
+      - foo-b
+      - foo-c
+```
+
+then PackageVariant resources will be created with these repository and package
+names:
+
+| Repository | Package Name |
+| ---------- | ------------ |
+| cluster-01 | foo          |
+| cluster-03 | foo          |
+| cluster-04 | foo          |
+| cluster-02 | foo-a        |
+| cluster-02 | foo-b        |
+| cluster-02 | foo-c        |
+| cluster-04 | foo-a        |
+| cluster-04 | foo-b        |
+| cluster-04 | foo-c        |
+
+Finally, the third possibility allows the use of *arbitrary* resources in the
+Porch cluster as targeting criteria. The `objectSelector` looks like this:
+
+```yaml
+spec:
+  upstream:
+    repo: example-repo
+    package: foo
+    revision: v1
+  targets:
+  - objectSelector:
+      apiVersion: krm-platform.bigco.com/v1
+      kind: Team
+      matchLabels:
+        org: hr
+        role: dev
+```
+
+It works exactly like the repository selector - in fact the repository selector
+is equivalent to the object selector with the `apiVersion` and `kind` values set
+to point to Porch Repository resources. That is, the repository name comes from
+the object name, and the package names come from the listed package names. In
+the description of the template, we will see how to derive different repository
+names from the objects.
+
+#### PackageVariant Template
+
+As previously discussed, the list entries generated by the target criteria
+result in PackageVariant entries. If no template is specified, then
+PackageVariant default are used, along with the downstream repository name and
+package name as described in the previous section. The template allows the user
+to have control over all of the values in the resulting PackageVariant. The
+template API is shown below.
+
+```yaml
```

Comment:
Nit: This should be `go` rather than `yaml`. It will give correct color coding and make it easier to read.